### ISSUE TITLE
tree: load job nodes on request

### ISSUE THREAD

**Issue by oliver-sanders** [cylc/cylc-ui#347]
When you click on a job in the Tree View you get a grey box with additional job information.

I think at present this information is part of the subscription and part of the tree.

We could consider changing this to a system where the data is dynamically loaded on-request (if not already in the store for some other reason) via a GraphQL query sent over WS. And perhaps spinning up the tree node for the job on request and destroying it after use.

This would prevent requesting excess job information when not displayed and reduce the node count in the tree view.

Questions:

* [ ] Are these job info boxes currently nodes in the tree view (or am I barking up the wrong tree).
* [ ] Does this make sense?

**Pull requests welcome!**

**Comment by kinow** [comment]
> Are these job info boxes currently nodes in the tree view (or am I barking up the wrong tree).

Not at the moment, but I had to make them into nodes for #330 . They are created with information returned from the GraphQL query, that part

```
{
  ...
  jobs {
    id 
    host
    jobId
    ...
  }
}
```

>Does this make sense?

I'm +1 to explore this idea. At the moment you will have cycle points, families, and tasks in the DOM. But we are still spending time in the JS scripting task, massaging the data - including jobs and job details - even though not everything is displayed.

This could be a good improvement, discarding both jobs and job nodes, and loading them on request - not sure if jobs could go away, however, as they "exist" kinda, as we list all the jobs when task is collapsed, but I think your new design document has a proposal to list only the most recent, another good improvement.

And, luckily, `infinite-tree` component supports dynamically loading nodes too :tada:

**Comment by oliver-sanders** [comment]
> not sure if jobs could go away

Yeah I don't think they will go away but hopefully we can avoid storing data like "started_time" for each job where not necessary.

For massive suites we could have ~10'000 tasks so the scaling factor is reasonable.
